gateway/ln-gateway/src/lib.rs

```diff
@@ -670,8 +672,13 @@ impl Gateway {
             address,
             federation_id,
         } = payload;
-
         let client = self.select_client(federation_id).await?;
+
+        let amount: bitcoin::Amount = match amount {
+            AmountOrAll::All => bitcoin::Amount::from_sat(client.get_balance().await.msats * 1000),
```

Comment:
I think this should probably use like `client.get_balance() - client.get_peg_out_fees()` and make sure this is valid amount